FutureNorthants/northants-website
# Feature/revalidate backend

To be used with https://github.com/FutureNorthants/northants-website/pull/326

This sends a request to the frontend when a node is saved. The request contains an array of paths that need to be revalidated by the frontend so they are updated. This happens before the request is sent to cloudflare to purge the cached paths. 

## To Test
- Checkout this branch, then edit the docker-compose.yml and update `REVALIDATE_LOCAL_PATH: "http://localhost:3000/"` to point to your IP address
- Run `make` to restart the docker containers and import the updated config with the new module
- Edit and publish a page and a request should be sent to the frontend. If the frontend is not running or unreachable, the exception should be caught and logged in Drupal

## Frontend Testing
- Checkout the frontend branch above and set the `REVALIDATE_API_TOKEN` in the .env.local file to match your token in the docker-compose.yml file
- Run `yarn install && yarn build && yarn start`
- Visit a service landing page on the frontend so it is made into a static page. Visit a child service page (that has the service page title in the breadcrumbs)
- Edit the title on the service landing page in the CMS and publish it. 
- Refresh the page in the frontend and it should be updated
- Visit the child page again and ensure the breadcrumb is also updated

## To work on
- It may not pickup changes to parent pages, such as the homepage, when a service page title is updated. This seems to be an issue with cloudflare cache purging specific urls too. I assume these are not changed very often. 
- It's not ideal having to edit the variables in the docker-compose.yml file. The .env file settings don't seem to overwrite the variables in the docker-compose.yml file.
